uhabits-android/src/main/java/org/isoron/uhabits/activities/habits/list/views/ButtonPanelView.kt

```diff
@@ -54,6 +55,12 @@ abstract class ButtonPanelView<T : View>(
 
     @Synchronized
     protected fun inflateButtons() {
+        if (preferences.isMidnightDelayEnabled) {
+            setStartDayOffset(3, 0)
+        } else {
+            setStartDayOffset(0, 0)
+        }
+
```

Comment:
Is this required because the previous code only runs during startup? I think if we need this here, then we almost certainly need it in many other places (for example, in `HeaderView`). I guess this feature is more tricky than I expected. Perhaps we should force an app restart after the user changes this preference.

Reply:
Ok then I think https://github.com/iSoron/uhabits/pull/739 might be more reasonable.